zrml/prediction-markets/src/lib.rs

```diff
@@ -203,6 +204,7 @@ mod pallet {
                 T::MarketCommons::remove_market_pool(&market_id)?;
             }
 
+            Self::clear_auto_open(&market_id)?;
             Self::clear_auto_close(&market_id)?;
```

Comment:
Yeah, great suggestion.

I tried this, and unfortunately, I ran into some trouble. You'll probably want to keep `clear_auto_close` and `clear_auto_open`, but let them call a function `clear_auto_cache` which is structured similarly to the `market_status_manager`. Otherwise, you'd have to keep writing `clear_auto_cache::<MarketIdsPerOpenBlock, MarketIdsPerOpenTimestamp>` where `clear_auto_close` used to be.

But then you must also specify whether or not to access `range.start` or `range.end` in `clear_auto_cache`. I don't think this has a good solution. You can either use a macro, or some kind of a convoluted closure construction. Either way, you end up with 30 lines of complex code to prevent 30 lines of code duplications.